CMakeLists.txt

```diff
@@ -0,0 +1,342 @@
+cmake_minimum_required(VERSION 3.28)
+
+cmake_policy(SET CMP0048 NEW)
+cmake_policy(SET CMP0091 NEW)
+
+set(PKG_VERSION 1.0.31)
+set(LIB_VERSION 1.74.0)
+
+project(tkrzw
+  VERSION ${PKG_VERSION}
+  DESCRIPTION "a set of implementations of DBM"
+  HOMEPAGE_URL "https://dbmx.net/tkrzw/"
+  LANGUAGES C CXX
+)
+add_library(tkrzw)
+set_target_properties(tkrzw PROPERTIES VERSION ${LIB_VERSION} SOVERSION 1)
+
+include(CMakePackageConfigHelpers)
+include(CPack)
+include(GNUInstallDirs)
+include(CMakeDependentOption)
+
+option(WITH_TOOLS "Build cli frontends" OFF)
+cmake_dependent_option(WITH_TESTS "Generate a test target" OFF "WITH_TOOLS" OFF)
```

Comment:
CMake has hidden feature that groups names based on prefix (Used by CMake GUI, Qt creator, VS2022...)

This should just be `TKRZW_TOOLS`, `TKRZW_TESTS`. 

They are grouped into `WITH`. I also used WITH in another two projects, and they cannot fix it anymore 😅 

<img width="300" src="https://github.com/user-attachments/assets/3bd8b7bc-18ad-4f95-a883-19c666a5ea58">

Reply:
Indeed, a `TKRZW_` prefix scales better then a `WITH_` prefix does.